src/search_engines.js

```diff
@@ -444,6 +444,79 @@ class GoogleSearch {
     }
     return false;
   }
+
+  changeImageSize(size) {
+    const openTool = document.querySelector(
+        '[class="PNyWAd ZXJQ7c"][jsname="I4bIT"]');
+    if (openTool != null) {
+      openTool.click();
+    }
+    const openSizeDropDown = document.querySelector(
+        '[class="xFo9P r9PaP"][jsname="wLFV5d"][aria-label="Size"]');
+    if (openSizeDropDown != null) {
+      openSizeDropDown.click();
+    }
+    const large = document.querySelector(
+        '[class="MfLWbb"][aria-label="Large"]');
+    const medium = document.querySelector(
+        '[class="MfLWbb"][aria-label="Medium"]');
+    const icon = document.querySelector(
+        '[class="MfLWbb"][aria-label="Icon"]');
+    const dropDownWithSize = document.querySelector(
+        '[class="xFo9P r9PaP Fmo8N"][jsname="wLFV5d"]');
+    if (large != null && medium != null && icon != null) {
```

Comment:
It seems possible to simplify the two switch statements. Not sure about the best way to do it, but maybe something like:

```javascript
switch(size) {
  case LARGE:
    if (dropDownWithSize != null) {
      if (dropDownWithSize.innerHTML == 'Large') {
        return;
      }
      if (large == null) {
        dropDownWithSize.click();
        // large = ...
      }
    }
    if (large != null) {
      large.click();
    }
  // ...
}
```

Now basically some of the code in the case can be extracted out to a function, say something like:
```javascript
const setImageSize = (dropDownWithSize, getButton) => {
  let button = getButton();
  if (dropDownWithSize != null && button == null) {
    dropDownWithSize.click();
    button = getButton();
  }
  if (button != null) {
    button.click();
  }
};

// ...


switch(size) {
  case LARGE:
    if (dropDownWithSize != null && dropDownWithSize.innerHTML == 'Large') {
      return;
    }
    setImageSize(dropDownWithSize, () => { 
      // return document...;
    });
    break;
  // ...
}

```